include/aspect/material_model/rheology/diffusion_dislocation.h

```diff
@@ -90,6 +90,33 @@ namespace aspect
                                             const double temperature,
                                             const SymmetricTensor<2,dim> &strain_rate) const;
 
+          /**
+           * Compute the derivative of the natural logarithm of the strain rate
+           * with respect to the natural logarithm of the stress
+           * at the current evaluation point (log stress) estimated by KINSOL.
+           */
+          void
+          compute_log_strain_rate_deriv(const Vector<double> &evaluation_point,
+                                        double pressure,
+                                        double temperature,
+                                        const Rheology::DiffusionCreepParameters diffusion_creep_parameters,
+                                        const Rheology::DislocationCreepParameters dislocation_creep_parameters,
+                                        double &log_strain_rate_deriv) const;
```

Comment:
All but one of these arguments are input arguments. Make them `const`. Also in the next function and probably in the implementations of these functions.

The last argument here is an output argument. Is it also an input argument? I generally think that these are difficult to read. My preference would be if the function *returned* what it computes.

Reply:
I think KINSOL requires that this function (which is associated with `nonlinear_solver.setup_jacobian`) does not return anything, but it must update the Jacobian log_strain_rate_deriv which is then used in `nonlinear_solver.solve_with_jacobian`.

Reply:
That's true for the lambda function from which you call this function, but *this* function does not have to satisfy the requirement.